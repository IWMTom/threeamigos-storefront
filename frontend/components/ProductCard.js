import { Link } from "../routes";
import formatMoney from "../lib/formatMoney";

const ProductCard = ({ product }) => (
	<div className="pure-u-1-3">
		<Link route="product" params={{ slug: product.slug }}>
			<a className={product.sale ? "card product sale" : "card product"}>
				{product.sale && <span className="product-sale">Sale 50%</span>}
				<img src={product.image_url} alt={product.name} />
				<span id="name">{product.name}</span>
				<h3 id="price">{formatMoney(product.price)}</h3>
			</a>
		</Link>
	</div>
);

export default ProductCard;
