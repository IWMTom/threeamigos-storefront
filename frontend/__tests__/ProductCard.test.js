import ProductCard from "../components/ProductCard";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

const mockedProduct = {
	id: "abcdefghijklmnopqrst",
	slug: "abc-def",
	sale: true,
	name: "Abc Def",
	image_url: "http://placehold.it/100x100",
	price: "199"
};

describe("<ProductCard />", () => {
	it("renders and matches the snapshot", () => {
		const wrapper = shallow(<ProductCard product={mockedProduct} />);

		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	// it("renders and displays correctly", () => {
	// 	const wrapper = shallow(<ProductCard product={mockedProduct} />);
	// 	const image = wrapper.find("img");
	// 	const name = wrapper.find("span#name");
	// 	const price = wrapper.find("h3#price");

	// 	expect(image.props().src).toBe(mockedProduct.image_url);
	// 	expect(name.text()).toBe(mockedProduct.name);
	// 	expect(price.text()).toBe("Â£1.99");
	// 	expect(wrapper.find("span.product-sale").exists()).toBe(true);
	// });
});
