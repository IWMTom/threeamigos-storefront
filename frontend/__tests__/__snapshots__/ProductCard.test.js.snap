// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> renders and matches the snapshot 1`] = `
<div
  className="pure-u-1-3"
>
  <LinkRoutes
    params={
      Object {
        "slug": "abc-def",
      }
    }
    route="product"
  >
    <a
      className="card product sale"
    >
      <span
        className="product-sale"
      >
        Sale 50%
      </span>
      <img
        alt="Abc Def"
        src="http://placehold.it/100x100"
      />
      <span
        id="name"
      >
        Abc Def
      </span>
      <h3
        id="price"
      >
        Â£1.99
      </h3>
    </a>
  </LinkRoutes>
</div>
`;
